File: DiscTypescript/src/Teste.ts
function main(): void {
    const readlineSync = require('readline-sync');
  
    const quantidadePastas = readlineSync.questionInt('Digite a quantidade de pastas que deseja criar: ');
    console.log();
  
    const pastas: Pasta[] = [];
    for (let i = 0; i < quantidadePastas; i++) {
      console.log(`-----------------------------------PASTA ${i + 1}-----------------------------------`);
      const nomePasta = readlineSync.question('Digite o nome da pasta: ');
  
      const novaPasta = new Pasta(nomePasta);
      pastas.push(novaPasta);
  
      const quantidadeSubpastas = readlineSync.questionInt(`Digite a quantidade de subpastas que deseja criar em '${novaPasta.nome}': `);
  
      if (quantidadeSubpastas > 0) {
        criarSubpastas(readlineSync, novaPasta, quantidadeSubpastas);
      } else {
        const quantidadeArquivos = readlineSync.questionInt(`Digite a quantidade de arquivos que deseja adicionar à pasta '${novaPasta.nome}': `);
        criarArquivos(readlineSync, novaPasta, quantidadeArquivos);
      }
  
      const tamanhoPasta = novaPasta.calcularTamanhoTotal();
      console.log(`Tamanho da pasta '${novaPasta.nome}': ${tamanhoPasta}`);
      console.log();
    }
  
    let opcaoExcluir = readlineSync.question('Deseja excluir as pastas criadas? (S/N): ');
    while (!(opcaoExcluir.toUpperCase() === 'S') && !(opcaoExcluir.toUpperCase() === 'N')) {
      console.log('Opção inválida! Deseja excluir as pastas criadas? (S/N): ');
      opcaoExcluir = readlineSync.question();
    }
  
    if (opcaoExcluir.toUpperCase() === 'S') {
      for (const pasta of pastas) {
        pasta.excluir();
      }
      pastas.length = 0;
      console.log('Pastas excluídas.');
    }
  }
  
  function criarSubpastas(readlineSync: any, pastaPai: Pasta, quantidadeSubpastas: number): void {
    for (let i = 0; i < quantidadeSubpastas; i++) {
      console.log();
      const nomeSubpasta = readlineSync.question('Digite o nome da subpasta: ');
  
      const novaSubpasta = new Pasta(nomeSubpasta);
      pastaPai.adicionarSubpasta(novaSubpasta);
  
      const quantidadeArquivos = readlineSync.questionInt('Digite a quantidade de arquivos que deseja adicionar à subpasta: ');
  
      criarArquivos(readlineSync, novaSubpasta, quantidadeArquivos);
    }
  }
  
  function criarArquivos(readlineSync: any, pasta: Pasta, quantidadeArquivos: number): void {
    for (let i = 0; i < quantidadeArquivos; i++) {
      const nomeArquivo = readlineSync.question('Digite o nome do arquivo: ');
      const tipoArquivo = readlineSync.question('Digite o tipo do arquivo: ');
      const tamanhoArquivo = readlineSync.questionInt('Digite o tamanho do arquivo: ');
  
      const novoArquivo = new Arquivo(nomeArquivo, tipoArquivo, tamanhoArquivo);
      pasta.adicionarArquivo(novoArquivo);
    }
  }
  
  main();

File: DiscJava/DiscoVirtual/src/br/com/disco/teste/Teste.java
package br.com.disco.teste;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

import br.com.disco.arquivo.Arquivo;
import br.com.disco.pasta.Pasta;

public class Teste {

	public static void main(String[] args) { 
        Scanner scanner = new Scanner(System.in);

        System.out.print("Digite a quantidade de pastas que deseja criar: ");
        int quantidadePastas = scanner.nextInt();
        System.out.println();

        List<Pasta> pastas = new ArrayList<>();
        for (int i = 0; i < quantidadePastas; i++) {
            System.out.println("-----------------------------------PASTA " + (i + 1) + "-----------------------------------");
            System.out.print("Digite o nome da pasta: ");
            String nomePasta = scanner.next();

            Pasta novaPasta = new Pasta(nomePasta);
            pastas.add(novaPasta);

            System.out.print("Digite a quantidade de subpastas que deseja criar em '" + novaPasta.getNome() + "': ");
            int quantidadeSubpastas = scanner.nextInt();

            if (quantidadeSubpastas > 0) {
                criarSubpastas(scanner, novaPasta, quantidadeSubpastas);
            } else {
                System.out.print("Digite a quantidade de arquivos que deseja adicionar à pasta '" + novaPasta.getNome() + "': ");
                int quantidadeArquivos = scanner.nextInt();
                criarArquivos(scanner, novaPasta, quantidadeArquivos);
            }

            int tamanhoPasta = novaPasta.calcularTamanhoTotal();
            System.out.println("Tamanho da pasta '" + novaPasta.getNome() + "': " + tamanhoPasta);
            System.out.println();
        }

        System.out.print("Deseja excluir as pastas criadas? (S/N): ");
        String opcaoExcluir = scanner.next();
        while(!(opcaoExcluir.equalsIgnoreCase("S")) && !(opcaoExcluir.equalsIgnoreCase("N"))) {
        	System.out.println("Opção inválida! Deseja excluir as pastas criadas? (S/N): ");
        	opcaoExcluir = scanner.next();
        }

        if (opcaoExcluir.equalsIgnoreCase("S")) {
            for (Pasta pasta : pastas) {
                pasta.excluir();
            }
            pastas.clear();
            System.out.println("Pastas excluídas.");
        }

        scanner.close();
    }

    private static void criarSubpastas(Scanner scanner, Pasta pastaPai, int quantidadeSubpastas) {
        for (int i = 0; i < quantidadeSubpastas; i++) {
            System.out.println();
            System.out.print("Digite o nome da subpasta: ");
            String nomeSubpasta = scanner.next();

            Pasta novaSubpasta = new Pasta(nomeSubpasta);
            pastaPai.adicionarSubpasta(novaSubpasta);

            System.out.print("Digite a quantidade de arquivos que deseja adicionar à subpasta: ");
            int quantidadeArquivos = scanner.nextInt();

            criarArquivos(scanner, novaSubpasta, quantidadeArquivos);
        }
    }

    private static void criarArquivos(Scanner scanner, Pasta pasta, int quantidadeArquivos) {
        for (int i = 0; i < quantidadeArquivos; i++) {
            System.out.print("Digite o nome do arquivo: ");
            String nomeArquivo = scanner.next();

            System.out.print("Digite o tipo do arquivo: ");
            String tipoArquivo = scanner.next();

            System.out.print("Digite o tamanho do arquivo: ");
            int tamanhoArquivo = scanner.nextInt();

            Arquivo novoArquivo = new Arquivo(nomeArquivo, tipoArquivo, tamanhoArquivo);
            pasta.adicionarArquivo(novoArquivo);
        }
    }
    }

	


